plugins/check_curl.c

```diff
@@ -1026,23 +1055,60 @@ check_http (void)
 
   if (strlen (header_expect)) {
     if (!strstr (header_buf.buf, header_expect)) {
+
       strncpy(&output_header_search[0],header_expect,sizeof(output_header_search));
+
       if(output_header_search[sizeof(output_header_search)-1]!='\0') {
         bcopy("...",&output_header_search[sizeof(output_header_search)-4],4);
       }
-      snprintf (msg, DEFAULT_BUFFER_SIZE, _("%sheader '%s' not found on '%s://%s:%d%s', "), msg, output_header_search, use_ssl ? "https" : "http", host_name ? host_name : server_address, server_port, server_url);
-      result = STATE_CRITICAL;
+
```

Comment:
Because the compiler was unhappy about using the same buffer to read AND write (which could result in overlap and weird results).
Allocating a buffer on the heap is probably the wrong approch. Using the stack might be better.

Reply:
Fixed it